In [1]:
import sys
sys.path.append('../')
import utils.data_preprocess as preprocess

In [2]:
pre = preprocess.preprocess("D:/DataSet/CICIDS2017/CSV/Wednesday-workingHours.pcap_ISCX.csv", labelColumnName='Label', depth=2)
X_train, X_test, y_train, y_test = pre.preprocessing()

In [3]:
from sklearn.tree import DecisionTreeClassifier
classifier = DecisionTreeClassifier(criterion='entropy',random_state=0)
classifier.fit(X_train, y_train.argmax(axis=1))

DecisionTreeClassifier(criterion='entropy', random_state=0)

In [4]:
y_pred = classifier.predict(X_test)

In [5]:
from sklearn.metrics import confusion_matrix
confusion_matrix(y_test.argmax(axis=1), y_pred)

array([[131578,     63],
       [    62,  75719]], dtype=int64)

In [6]:
from sklearn import metrics
print("F1 Score : ", metrics.f1_score(y_test.argmax(axis=1), y_pred, average='binary'))
print("Precision : ", metrics.precision_score(y_test.argmax(axis=1), y_pred, average='binary'))
print("Recall : ", metrics.recall_score(y_test.argmax(axis=1), y_pred, average='binary'))

F1 Score :  0.9991752604527491
Precision :  0.9991686680214299
Recall :  0.9991818529710613


### gini
```
array([[131506,     82],
       [    66,  75768]], dtype=int64)
```
F1 Score :  0.9990242873341948
Precision :  0.9989189189189189
Recall :  0.999129677980853
### entropy
```
array([[131578,     63],
       [    62,  75719]], dtype=int64)
```
F1 Score :  0.9991752604527491
Precision :  0.9991686680214299
Recall :  0.9991818529710613